FeedAPIChallenge/Feed API/FeedImagesMapper.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  FeedImagesMapper.swift
+//  FeedAPIChallenge
+//
+//  Created by Bryan Hoke on 6/21/21.
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+final class FeedImagesMapper {
+	private struct Root: Decodable {
+		let items: [Item]
+
+		var feed: [FeedImage] {
+			items.map(\.image)
+		}
+	}
+
+	private struct Item: Decodable {
+		let id: UUID
+		let description: String?
+		let location: String?
+		let url: URL
+
+		var image: FeedImage {
+			FeedImage(id: id, description: description, location: location, url: url)
+		}
+
+		enum CodingKeys: String, CodingKey {
+			case id = "image_id"
+			case description = "image_desc"
+			case location = "image_loc"
+			case url = "image_url"
+		}
+	}
```

Comment:
The `Item` properties can be named exactly as the Feed API specs/contract. This way, the code is shorter and more concise since it doesn't need the `CodingKeys` enum.

Using underscores is not the convention in Swift, but in this case, this struct exists to describe remote models with different conventions. So it is ok that the remote model's properties don't follow swift's convention - it follows the remote model's convention!

Following the remote model's convention here clarifies the intent of the code and makes maintenance easier (especially when you have many remote models to represent in the app!)

```suggestion
	private struct Item: Decodable {
		let image_id: UUID
		let image_desc: String?
		let image_loc: String?
		let image_url: URL

		var image: FeedImage {
			FeedImage(id: image_id, description: image_desc, location: image_loc, url: image_url)
		}
	}
```